### PR TITLE
Add a single callback function to handle all tool calls

### PR DESCRIPTION
**Pull request by chadbailey59**

### CODE REVIEW

File: src/pipecat/services/ai_services.py [hunk 1]
```diff
@@ -115,28 +115,38 @@ def __init__(self, **kwargs):
         self._start_callbacks = {}
 
     # TODO-CB: callback function type
-    def register_function(self, function_name: str, callback, start_callback=None):
+    def register_function(self, function_name: str | None, callback, start_callback=None):
+        if not function_name: # then use this callback for all functions
+            function_name = "*"
```
Comment (aconchillo) [inline src/pipecat/services/ai_services.py:120]: No need to use `*`, in Python you can use `None` keys. So, no need to do anything special.

File: src/pipecat/services/ai_services.py [hunk 2]
```diff
@@ -115,28 +115,38 @@ def __init__(self, **kwargs):
...
+            function_name = "*" 
         self._callbacks[function_name] = callback
         if start_callback:
             self._start_callbacks[function_name] = start_callback
+        
 
-    def unregister_function(self, function_name: str):
+    def unregister_function(self, function_name: str | None):
+        if not function_name: # then use this callback for all functions
+            function_name = "*"
```
Comment (aconchillo) [inline src/pipecat/services/ai_services.py:128]: Same here. Just use `None` as a key.

File: src/pipecat/services/ai_services.py [hunk 3]
```diff
@@ -115,28 +115,38 @@ def __init__(self, **kwargs):
...
         del self._callbacks[function_name]
         if self._start_callbacks[function_name]:
             del self._start_callbacks[function_name]
 
     def has_function(self, function_name: str):
+        if "*" in self._callbacks.keys():
+            return True
```
Comment (aconchillo) [inline src/pipecat/services/ai_services.py:135]: Same here. No need for this.

File: src/pipecat/services/ai_services.py [hunk 4]
```diff
@@ -115,28 +115,38 @@ def __init__(self, **kwargs):
...
         return function_name in self._callbacks.keys()
 
     async def call_function(self, function_name: str, args):
         if function_name in self._callbacks.keys():
             return await self._callbacks[function_name](self, args)
+        elif "*" in self._callbacks.keys():
+            return await self._callbacks["*"](self, function_name, args)
```
Comment (aconchillo) [inline src/pipecat/services/ai_services.py:142]: And no need for this either.

File: src/pipecat/services/ai_services.py [hunk 5]
```diff
@@ -115,28 +115,38 @@ def __init__(self, **kwargs):
...
         return None
 
     async def call_start_function(self, function_name: str):
         if function_name in self._start_callbacks.keys():
             await self._start_callbacks[function_name](self)
-
+        elif "*" in self._callbacks.keys():
+            return await self._start_callbacks["*"](self, function_name)
```
Comment (aconchillo) [inline src/pipecat/services/ai_services.py:149]: Same.

**Comment by chadbailey59** [comment]
I've got too many branches and I'm getting myself confused, so I moved this change to my other function calling PR

**Pull request closed without merging**